### PR TITLE
Adding CTAs to articles and a blog post.

### PR DESCRIPTION
**Pull request by bradmccarty**

### CODE REVIEW

File: astro/src/content/articles/tokens/revoking-jwts.md [hunk 1]
```diff
@@ -29,6 +29,8 @@ The ToDo Backend in the diagram can use the JWT and the public key to verify the
```
Comment (mooreds) [inline astro/src/content/articles/tokens/revoking-jwts.md:29]: We should really consider updating the diagram above (`jwt-revoke_350.png`) if this is a high traffic page.

File: astro/src/content/articles/tokens/jwt-components-explained.md [hunk 2]
```diff
@@ -77,6 +77,8 @@ This table is drawn from RFC 7518. As only HS256 is required to be compliant wit
 
 Other metadata is also stored in this part of the JWT. The `typ` header indicates the type of the JWT. In this case, the value is `JWT`, but other values are valid. For instance, if the JWT conforms to RFC 9068, it may have the value `at+JWT` indicating it is an access token.
 
+[![A call to action image displaying a link to download this article as an ebook](/img/cta/jwt-guide-cta.png 'Download this article as an ebook')](/ebooks/breaking-down-json-web-tokens?utm_medium=cta&utm_source=articles&utm_campaign=jwt_ebook)
```
Comment (mooreds) [inline astro/src/content/articles/tokens/jwt-components-explained.md:80]:
```suggestion
[![A call to action image displaying a link to download an ebook on this topic.](/img/cta/jwt-guide-cta.png 'Download this article as an ebook')](/ebooks/breaking-down-json-web-tokens?utm_medium=cta&utm_source=articles&utm_campaign=jwt_ebook)
```

File: astro/src/content/articles/tokens/revoking-jwts.md [hunk 3]
```diff
@@ -29,6 +29,8 @@ The ToDo Backend in the diagram can use the JWT and the public key to verify the
 
 The most common solution is to reduce the duration of the JWT and revoke the refresh token so that the user can't generate a new JWT. With this setup, the JWT's expiration duration is set to something short (5-10 minutes) and the refresh token is set to something long (2 weeks or 2 months). At any time, an administrator can revoke the refresh token which means that the user must re-authenticate to get a new JWT. That is unless they happen to have a valid JWT.
 
+[![A call to action image displaying a link to download this article as an ebook](/img/cta/jwt-guide-cta.png 'Download this article as an ebook')](/ebooks/breaking-down-json-web-tokens?utm_medium=cta&utm_source=articles&utm_campaign=jwt_ebook)
```
Comment (mooreds) [inline astro/src/content/articles/tokens/revoking-jwts.md:32]:
```suggestion
[![A call to action image displaying a link to download an ebook on this topic.](/img/cta/jwt-guide-cta.png 'Download this article as an ebook')](/ebooks/breaking-down-json-web-tokens?utm_medium=cta&utm_source=articles&utm_campaign=jwt_ebook)
```

**Review by mooreds — CHANGES_REQUESTED** [review]
A few suggested changes and one question.

File: astro/src/content/articles/tokens/revoking-jwts.md (hunk 1 above)
Comment (bradmccarty) [reply astro/src/content/articles/tokens/revoking-jwts.md:29]: Yes! https://linear.app/fusionauth/issue/DES-560/article-or-revoking-jwts-or-articles

File: astro/src/content/blog/how-to-securely-implement-oauth-angular.mdx [hunk 4]
```diff
@@ -32,6 +32,8 @@ You need to have the following software installed before you begin:
 
 You'll also want to make sure your system meets the [memory, storage and CPU requirements](/docs/get-started/download-and-install/system-requirements) for FusionAuth.
 
+[![A call to action image displaying a link to download this article as an ebook](/img/cta/auth-react-guide-cta.png 'Download this article as an ebook')](/tech-papers/use-oauth-to-add-authentication-to-your-react-app?utm_medium=cta&utm_source=articles&utm_campaign=oauth_ebook)
```
[the remark this answers is not in the document]
Comment (bradmccarty) [reply astro/src/content/blog/how-to-securely-implement-oauth-angular.mdx:35]: The React on Angular is known and intentional (I disagree with it, but this is a test). 

I'll change the UTM. Thanks!
```suggestion
[![A call to action image displaying a link to download this article as an ebook](/img/cta/auth-react-guide-cta.png 'Download this article as an ebook')](/tech-papers/use-oauth-to-add-authentication-to-your-react-app?utm_medium=cta&utm_source=articles&utm_campaign=angular_ebook)
```

File: astro/src/content/blog/how-to-securely-implement-oauth-angular.mdx [hunk 5]
```diff
@@ -32,6 +32,8 @@ You need to have the following software installed before you begin:
 
 You'll also want to make sure your system meets the [memory, storage and CPU requirements](/docs/get-started/download-and-install/system-requirements) for FusionAuth.
 
+[![A call to action image displaying a link to download this article as an ebook](/img/cta/auth-react-guide-cta.png 'Download this article as an ebook')](/tech-papers/use-oauth-to-add-authentication-to-your-react-app?utm_medium=cta&utm_source=articles&utm_campaign=angular_ebook)
```
Comment (mooreds) [inline astro/src/content/blog/how-to-securely-implement-oauth-angular.mdx:35]: Heya, why is there a react CTA on an angular page?

File: astro/src/content/articles/oauth/complete-list-oauth-grants.md [hunk 6]
```diff
@@ -20,6 +20,8 @@ This article will go through all the different OAuth 2 grant types and explain t
 
 If you aren't familiar with terms like Authorization Server and Resource Server, you might be interested in [What is OAuth](/articles/oauth/what-is-oauth/).
 
+[![A call to action image displaying a link to download this article as an ebook](/img/cta/oauth-grants-guide-cta.png 'Download this article as an ebook')](/tech-papers/complete-list-of-oauth-2-grants?utm_medium=cta&utm_source=articles&utm_campaign=oauth2-grants)
```
Comment (mooreds) [inline astro/src/content/articles/oauth/complete-list-oauth-grants.md:23]:
```suggestion
[![A call to action image displaying a link to download this article as an ebook.](/img/cta/oauth-grants-guide-cta.png 'Download this article as an ebook')](/tech-papers/complete-list-of-oauth-2-grants?utm_medium=cta&utm_source=articles&utm_campaign=oauth2-grants)
```

**Review by mooreds — CHANGES_REQUESTED** [review]

**Review by mooreds — APPROVED** [review]

**Pull request merged**